pkg/network/node/networkpolicy.go

```diff
@@ -456,6 +456,13 @@ func (np *networkPolicyPlugin) parseNetworkPolicy(npns *npNamespace, policy *net
 					npp.watchesNamespaces = true
 					peerFlows = append(peerFlows, np.selectNamespaces(peer.NamespaceSelector)...)
 				}
+			} else if peer.IPBlock != nil {
+				// As per https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/networking/types.go#L181
+				if peer.NamespaceSelector != nil || peer.PodSelector != nil {
+					return nil, fmt.Errorf("PodSelector or NamespaceSeletor can not be used along with IPBlocks")
```

Comment:
You don't need to check this. The upstream validation code will already have ensured that